src/main/kotlin/app/nush/bot/commands/Nick.kt

```diff
@@ -0,0 +1,96 @@
+package app.nush.bot.commands
+
+import app.nush.bot.Config.Companion.config
+import com.jessecorbett.diskord.api.rest.CreateDM
+import com.jessecorbett.diskord.dsl.Bot
+import com.jessecorbett.diskord.dsl.CommandSet
+import com.jessecorbett.diskord.dsl.command
+import com.jessecorbett.diskord.util.authorId
+import com.jessecorbett.diskord.util.changeNickname
+import com.jessecorbett.diskord.util.sendMessage
+import com.jessecorbett.diskord.util.words
+
+object Nick : Command {
+    lateinit var bot: Bot
+    override fun init(bot: Bot, prefix: CommandSet) {
+        this.bot = bot
+        with(prefix) {
+            command("nick") {
+                request(
+                    bot,
+                    authorId,
+                    words.drop(1).joinToString(separator = " ")
+                )
+            }
+        }
+    }
+
+    suspend fun request(bot: Bot, userId: String, newName: String) {
+        with(bot) {
+            val guildClient = clientStore.guilds[config.guildId]
+            val channelClient = clientStore.channels[config.excoChannelId]
+            clientStore.channels[clientStore.discord.createDM(
+                CreateDM(
+                    userId
+                )
+            ).id].sendMessage("Your rename request has been sent to the exco, it will be processed as soon as possible.")
+            val renameMessage =
+                channelClient.sendMessage("${guildClient.getMember(userId).nickname}(<@!$userId>) has requested to be renamed to \"$newName\". ✅ if you want to accept the rename request, ❎ if you do not want to accept the rename request")
+            renameMessage.react("✅")
+            renameMessage.react("❎")
+            reactionAdded { messageReaction ->
+                try {
```

Comment:
What is this for?

Reply:
react to the message when it's sent, so exco only has to click, instead of selecting the reaction

Reply:
I mean the try/catch

Reply:
well basically if the message is not of the correct type some form of error will be thrown inside the try block then I just ignore the case by putting nothing in catch

Reply:
I think this is still unnecessary